packages/aws-cdk-lib/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -452,6 +472,33 @@ export class StateMachine extends StateMachineBase {
       }
     }
 
+    if (props?.kmsDataKeyReusePeriodSeconds && !isValidKmsDataKeyReusePeriodSeconds(props.kmsDataKeyReusePeriodSeconds)) {
+      throw new Error('kmsDataKeyReusePeriodSeconds needs to be a value between 60 and 900');
+    }
```

Comment:
If the customer doesn't provide a KMS could this have any effect? Perhaps we should invalidate this case?

Reply:
Makes sense will add that validation.